### PR TITLE
No longer using `compile-sql` to encode-sql-args lists.

### PR DESCRIPTION
**Pull request by danmason**
Similar to a137d41

### CODE REVIEW

File: core/src/main/clojure/xtdb/log.clj [hunk 1]
```diff
@@ -196,17 +196,14 @@
...
-(defn encode-sql-args [^BufferAllocator allocator, query, arg-rows]
-  (let [plan (sql/compile-query query)
-        {:keys [^long param-count]} (meta plan)
-
-        vecs (ArrayList. param-count)]
+(defn encode-sql-args [^BufferAllocator allocator, arg-rows]
+  (let [max-param-count (apply max (map count arg-rows))
+        vecs (ArrayList. max-param-count)]
     (try
-      ;; TODO check arg count in each row, handle error
```
Comment (danmason) [inline core/src/main/clojure/xtdb/log.clj:205]: If indeed we do see have a mismatch here expecting to throw an error at a later point when compiling the query.

Comment (jarohen) [reply core/src/main/clojure/xtdb/log.clj:205]: @danmason could you pop a test in to confirm this?

File: core/src/main/clojure/xtdb/log.clj [hunk 2]
```diff
@@ -196,17 +196,14 @@
       (.writeObject xtql-op-writer (ClojureForm. op))
       (.endStruct xtql-writer))))
 
-(defn encode-sql-args [^BufferAllocator allocator, query, arg-rows]
-  (let [plan (sql/compile-query query)
-        {:keys [^long param-count]} (meta plan)
-
-        vecs (ArrayList. param-count)]
+(defn encode-sql-args [^BufferAllocator allocator, arg-rows]
+  (let [max-param-count (apply max (map count arg-rows))
```
Comment (jarohen) [inline core/src/main/clojure/xtdb/log.clj]: `apply max` will fail here if `arg-rows` is empty - I assume that's handled elsewhere?

File: core/src/main/clojure/xtdb/log.clj [hunk 3]
```diff
@@ -226,7 +223,7 @@
         (.writeObject query-writer sql)
 
         (when-let [arg-rows (.argRows op)]
```
Comment (jarohen) [inline core/src/main/clojure/xtdb/log.clj]: `not-empty` call here around `argRows` would alleviate the `apply max` concern above

**Review by jarohen — APPROVED** [review]
couple of things to address but otherwise good to go, cheers @danmason :pray:

File: core/src/main/java/xtdb/vector/RelationReader.java [hunk 4]
```diff
@@ -33,6 +33,10 @@ public int rowCount() {
         return rowCount;
     }
 
+    public int argCount() {
```
Comment (jarohen) [inline core/src/main/java/xtdb/vector/RelationReader.java]: I wouldn't add this here - it's not something that makes sense for all RelationReaders

**Pull request merged**